benches/bloom_bench.rs

```diff
@@ -0,0 +1,64 @@
+use bytes::{BufMut, BytesMut};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use slatedb::filter::{filter_hash, BloomFilterBuilder};
+fn bloom_filter_insert_benchmark(c: &mut Criterion) {
+    let mut builder = BloomFilterBuilder::new(10); // Set bits_per_key
+    let key_sz = std::mem::size_of::<u32>();
+
+    c.bench_function("bloom_filter insert 10000000 keys", |b| {
+        b.iter(|| {
+            for i in 0..10000000 {
+                let mut bytes = BytesMut::with_capacity(key_sz);
```

Comment:
I don't think we should be generating the keys inside of the benchmark. We want the benchmark to give us some idea of the throughput of the code under test, which shouldn't include key generation which can be quite expensive. Having a more representative number will help us better understand the magnitude of any improvements we make. I'd suggest pre-generating these outside the bench method.